packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/listener/l7/create/controller.js

```diff
@@ -0,0 +1,73 @@
+export default class OctaviaLoadBalancerCreateL7PolicyCtrl {
+  /* @ngInject */
+  constructor(OctaviaLoadBalancerL7Service, Alerter, $translate) {
+    this.OctaviaLoadBalancerL7Service = OctaviaLoadBalancerL7Service;
+    this.Alerter = Alerter;
+    this.$translate = $translate;
+  }
+
+  $onInit() {
+    this.loading = false;
+    this.model = {
+      name: '',
+      position: 1,
+      action: undefined,
+      redirectHttpCode: undefined,
+      redirectPool: undefined,
+      redirectPrefix: undefined,
+      redirectUrl: undefined,
+    };
+  }
+
+  submit() {
+    this.loading = true;
+    this.trackL7CreatePolicyAction('confirm');
+    this.OctaviaLoadBalancerL7Service.createL7Policy(
+      this.projectId,
+      this.region,
+      this.listenerId,
+      this.model.name,
+      this.model.position,
+      this.model.action,
+      this.model.redirectHttpCode,
+      this.model.redirectPool?.id,
+      this.model.redirectPrefix,
+      this.model.redirectUrl,
```

Comment:
Why don't you use` this.model `as arguments instead ? Function parameters are difficult to read